networkx/algorithms/efficiency.py

```diff
@@ -0,0 +1,140 @@
+    --------
+    local_efficiency
+    global_efficiency
+
+    References
+    ----------
+    .. [1] Latora, Vito, and Massimo Marchiori.
+           "Efficient behavior of small-world networks."
+           *Physical Review Letters* 87.19 (2001): 198701.
+           http://dx.doi.org/10.1103/PhysRevLett.87.198701
+
+    """
+    return 1 / nx.shortest_path_length(G, u, v)
+
+
+@not_implemented_for('directed')
+def local_efficiency(G):
+    """Returns the average local efficiency of the graph.
+
+    The *efficiency* of a pair of nodes in a graph is the multiplicative
+    inverse of the shortest path distance between the nodes. The *local
+    efficiency* of a node in the graph is the average global efficiency of the
+    subgraph induced by the neighbors of the node. The *average local
+    efficiency* is the average of the local efficiencies of each node [1]_.
+
+    Parameters
+    ----------
+    G : :class:`networkx.Graph`
+        An undirected graph for which to compute the average local efficiency.
+
+    Returns
+    -------
+    float
+        The average local efficiency of the graph.
+
+    Notes
+    -----
+    Edge weights are ignored when computing the shortest path distances.
+
+    See also
+    --------
+    global_efficiency
+
+    References
+    ----------
+    .. [1] Latora, Vito, and Massimo Marchiori.
+           "Efficient behavior of small-world networks."
+           *Physical Review Letters* 87.19 (2001): 198701.
+           http://dx.doi.org/10.1103/PhysRevLett.87.198701
+
+    """
+    # TODO This summation can be trivially parallelized.
+    return sum(global_efficiency(G.subgraph(G[v])) for v in G) / len(G)
+
+
+@not_implemented_for('directed')
+def global_efficiency(G):
+    """Returns the average global efficiency of the graph.
+
+    The *efficiency* of a pair of nodes in a graph is the multiplicative
+    inverse of the shortest path distance between the nodes. The *average
+    global efficiency* of a graph is the average efficiency of all pairs of
+    nodes [1]_.
+
+    Parameters
+    ----------
+    G : :class:`networkx.Graph`
+        An undirected graph for which to compute the average global efficiency.
+
+    Returns
+    -------
+    float
+        The average global efficiency of the graph.
+
+    Notes
+    -----
+    Edge weights are ignored when computing the shortest path distances.
+
+    See also
+    --------
+    local_efficiency
+
+    References
+    ----------
+    .. [1] Latora, Vito, and Massimo Marchiori.
+           "Efficient behavior of small-world networks."
+           *Physical Review Letters* 87.19 (2001): 198701.
+           http://dx.doi.org/10.1103/PhysRevLett.87.198701
+
+    """
+    n = len(G)
+    denom = n * (n - 1)
+    # TODO This can be made more efficient by computing all pairs shortest
+    # path lengths in parallel.
+    #
+    # TODO This summation can be trivially parallelized.
+    return sum(efficiency(G, u, v) for u, v in combinations(G, 2)) / denom
```

Comment:
> I think they probably care about it since efficiency is defined between [0, 1]

Oh I see, so we should use `permutations` in the numerator and `n * (n - 1)` in the denominator. This allows the formula to work for directed graphs also (though they are currently disabled). I will change this to be permutations and add the tests you suggest.